src/main/java/org/jnode/fs/xfs/extent/BPlusTreeDataExtent.java

```diff
@@ -0,0 +1,220 @@
+package org.jnode.fs.xfs.extent;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+
+import org.jnode.fs.xfs.XfsObject;
+import org.jnode.fs.xfs.XfsFileSystem;
+import org.jnode.fs.xfs.XfsValidSignature;
+
+/**
+ *
+ * B+tree data extent
+ * Provides the infrastructure to read the b+tree data.
+ *
+ */
+public class BPlusTreeDataExtent extends XfsObject {
+
+    /**
+     * The magic number for a BMBT block (V5).
+     */
+    private final static Long MAGIC = asciiToHex("BMA3");
+
+    /**
+     * The level of the tree in which this block is found.
+     */
+    private final long level;
```

Comment:
While I think it's cleaner to declare these values, other areas (such as `Superblock`) just read the value from the underlying data buffer in the associated function - in this case,  the function `getLevel()` could just return `XfsObject.getUInt16(4)`. This would limit memory usage.
Just had a look where this is being called. Seems like the data buffer that's being passed into the constructor is being allocated (which I don't think is necessarily ideal). I'll have to see if there's better ways of dealing with this memory allocation.

Reply:
Code change applied. Thanks.